onnx/defs/tensor/defs.cc

```diff
@@ -3144,7 +3144,7 @@ ONNX_OPERATOR_SET_SCHEMA(
           auto* dim = output_shape.add_dim();
           if (hasInputShape(ctx, 0)) {
             const TensorShapeProto& input_shape = getInputShape(ctx, 0);
-            dim->set_dim_value(input_shape.dim_size());
+            dim->set_dim_value(input_shape.dim_size() == 0? 1 : input_shape.dim_size());
```

Comment:
I think the problem here is that the previous code is confusing the rank of the output-tensor and the dimensions of the tensor. The output is not 1-dimensional (based on the ORT implementation and numpy spec), since we return a tuple of indices for each non-zero value. So, the output is 2-dimensional, with one dimension-size being the input-tensor's rank and the other dimension having an unknown-value (== number of non-zero values in input).

This should be clarified better in the spec.

Reply:
As to what happens for a scalar: The numpy spec deprecates this since version 1.17, asking users to explicitly convert it to at-least 1D. This is sort of related to above issue as well, because the output will have one-dimension of size 0 (rank of input tensor), unless it is special-cased to be at-least 1. I don't think we need to special-case this (unless there is a good reason, that is some model depending on this.)